### ISSUE TITLE
Allow recovering head replicas to contribute to DVC

### ISSUE THREAD

**Issue by matklad** [tigerbeetle/tigerbeetle#1376]
We are a little less tolerant of storage faults if they cause a replica to enter into recovering head --- there, corruptions in _unrelated_ disk sectors might make the cluster available. This seems fixabe -- recovering head replicas can still contribute some nacks to DVC quorum, they just need to be extra careful

See comment at https://github.com/tigerbeetle/tigerbeetle/issues/1283#issuecomment-1864856466 for details. 

The TL;DR there is that it seems like CTRL protocol is actually a strict superset of view-change protocol. We use the fact that a replica is _not_ recovering its head to nack a whole bunch of slots implicitly, but we could make this nacking explicit.

**Comment by matklad** [comment]
Partially fixed by not getting replicas into recovering head in many cases!
